src/Server.cpp

```diff
@@ -656,6 +626,12 @@ void cServer::KickUser(int a_ClientID, const AString & a_Reason)
 
 void cServer::AuthenticateUser(int a_ClientID, const AString & a_Name, const AString & a_UUID, const Json::Value & a_Properties)
 {
+	if (GetNumPlayers() >= GetMaxPlayers())
```

Comment:
This still leaves potential for a race condition if multiple threads can execute AuthenticateUser. Is this function tied to a single thread? If not you can fix it by moving this condition to after taking the m_CSClients lock, so that only one thread can be between checking the condition and incrementing, but you should put a comment in here documenting that if you do so.